### PR TITLE
Adds send score framework and cleans up code and docs

### PR DESCRIPTION
**Pull request by sfreitas9**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,6 +97,32 @@ app.post('/project/return', (req, res) => {
...
+This library requires MongoDB.  If you do not currently have MongoDB setup, follow these instructions.
+
+* MacOS - use homebrew: https://docs.mongodb.com/master/tutorial/install-mongodb-on-os-x/
+* Windows - use the installer from here: https://docs.mongodb.com/master/tutorial/install-mongodb-on-windows/
+
+Once you have MongoDB setup, make note of your MongoDB URI, username, and password. You will need to set these up as environment variables either in your production environment or in your development .env file.  
+
+Add the following to your server.js file:
+```
+mongoose.connect(process.env.MONGODB_URI, {
```
Comment (GianFiyah) [inline README.md]: We should use a local Mongo DB instance for getting started: ie.
```
mongoose.connect('mongodb://localhost:27017/TESTLTI', {
  useNewUrlParser: true
},
  (err) => {
    if(err) {
      return console.log(err);
    }
});
mongoose.Promise = Promise;
```

File: README.md [hunk 2]
```diff
@@ -275,33 +302,35 @@ The Tool example will run on `http://localhost:3000/` in your browser.  The exam
...
-The Tool example provides a test suite to verify portions of the basic functionality of the Library.  
+The Library provides a test suite to verify portions of the basic functionality.  
 
 The tests require the use of a .env file, which if you are new to Node, is a a simple configuration text file that is used to pass customized variables into your application’s environment. 
 
-You will need to load the test data in your .env file for the follwoing.  You can use LMS Global's Reference Implementation Tool to generate keys (https://lti-ri.imsglobal.org/keygen/index).
-* CLIENT_ID=client12345
-* CLIENT_SECRET=<RSA Private Key>
-* CLIENT_PUBLIC=<Public Key>
+You will need to load the test data in your .env file for the following:
```
Comment (GianFiyah) [inline README.md]: For example tool purposes, database storage should occur locally using 'mongodb://localhost:27017/TESTLTI' in the mongoose.connect() function. Can mention that for production, they should implement a cloud storage service. 

NO NEED FOR .env in example tool.

File: README.md [hunk 3]
```diff
@@ -311,6 +340,8 @@ LMS - Learning Management System.  Referred to as Platforms in this document.
 
 OAuth 2.0 - LTI 1.3 specifies the use of the OAuth 2.0 Client Credential Grant mechanism to secure web services between trusted systems.  This Library makes use of JSON Web Tokens, JWT, for the access tokens. (https://www.imsglobal.org/spec/security/v1p0) 
 
+OIDC - OpenID Connect is a simple identity layer on top of the OAuth 2.0 protocol. It allows Clients to verify the identity of the End-User based on the authentication performed by an Authorization Server, as well as to obtain basic profile information about the End-User in an interoperable and REST-like manner.  https://openid.net/connect/
+
 Platform - previously referred to as the Tool Consumer is the Learning Management System (LMS) that the educational provider is using.  Examples are Moodle, Blackboard, Edgenuity, Canvas, Schoology, etc.
 
 Tool - previously referred to as the Tool Provider, this is the external Tool that contains educational material to include in a course.  Tools can range from a single piece of content, for example a quiz, to an extensive interactive website featuring specialized course content.
```
Comment (GianFiyah) [inline README.md]: Great job, Sherry!

File: README.md [hunk 4]
```diff
@@ -82,16 +64,29 @@ app.post('/oidc', (req, res) => {
...
+  if (!req.body.error) {
+    send_score(req, req.session.grade, < maximum grade allowed >);
+  } else {
+    res.status(401).send('Access denied: ' + req.params.error);
+  }
+});
+```
+
+*Within your Tool's route where grading is performed, set up the score to be returned to the Platform.*
+This initiates the score submittal, which will end at the above /auth_code route where the score is finally sent
```
[the remark this answers is not in the document]
Comment (sfreitas9) [reply README.md]: done

File: README.md [hunk 5]
```diff
@@ -141,31 +162,36 @@ registerPlatform(
...
+
 ```
+mongod
 npm start
 ```
-You need to access the generated Client Public key by making a GET request to `<your base URL>/publickey` (TODO:  verify endpoint when implemented).  This key must be put into the Tool's Public Key field in Step 4 above on the Platform.
+
+Now that your server is running, you are able to access the Tool's generated Client Public key by making a GET request to `<your base URL>/publickey` (TODO:  verify endpoint when implemented).  This key must be put into the Tool's Public Key field in Step 4 above on the Platform.  
 
 ---
```
[the remark this answers is not in the document]
Comment (sfreitas9) [reply README.md]: added

File: README.md (hunk 2 above)
Comment (sfreitas9) [reply README.md]: removed

File: README.md (hunk 3 above)
Comment (sfreitas9) [reply README.md]: Thanks for reading it in detail!

File: README.md [hunk 6]
```diff
@@ -111,25 +132,25 @@ Within the Platform, the Site Administrator needs to setup the External Tool.  F
...
 - Choose 'Always' Accept grades from the Tool, if Tool should send grades to Platform  
 
-After saving the External Tool, the Platform should assign a Client ID to the Tool and provide endpoints.  Make note of all of this information as it will be used in Step 4.  
+After saving the External Tool, the Platform will assign a Client ID to the Tool and provide endpoints.  Make note of all of this information as it will be used in Step 5.  
 
 ### 5. Register Platform with Tool
 
-In order register a Platform with the Tool, add a call to `registerPlatform` in your server file, with the values you received from Step 3.
+In order register a Platform with the Tool, add a call to `registerPlatform` in your server file, with the values you received from Step 4.
 ```
```
[the remark this answers is not in the document]
Comment (sfreitas9) [reply README.md]: done

**Pull request merged**